### PR TITLE
feat(daemon): auto-detect OpenClaude as a fallback for Claude Code

### LINKED ISSUE #235: Support OpenClaude
OpenClaude is a fork based on (leaked) Claude Code. The command line is no difference but change the name, but open-design cannot detect and choose it.
https://github.com/Gitlawb/openclaude

### PR DESCRIPTION
**Pull request by Sid-Qin**
## Summary

Adds a generic `fallbackBins` slot on agent definitions and uses it to make [OpenClaude](https://github.com/Gitlawb/openclaude) — an argv-compatible Claude Code fork — auto-detect as the **Claude Code** agent. Users with only `openclaude` on PATH no longer need to write a wrapper script.

`Closes #235`.

## Why

Today, agent detection in [`apps/daemon/src/agents.ts`](https://github.com/nexu-io/open-design/blob/main/apps/daemon/src/agents.ts) keys off a single binary name (`def.bin`):

```ts
async function probe(def) {
  const resolved = resolveOnPath(def.bin);
  if (!resolved) {
    return { ...stripFns(def), available: false, ... };
  }
  // …
}
```

OpenClaude ships as a Node-distributed CLI under the binary name `openclaude`, so on a `openclaude`-only install Claude Code is reported as "not detected" even though the CLI is fully argv-compatible. @GarryLai (issue author) confirmed in #235 that a wrapper script (`Path B`) works, then asked for the same to happen automatically. @lefarcen replied with two designs (fallback vs separate entry) and asked which to take. This PR implements the **fallback** approach the author suggested.

## Design choice — fallback vs separate AGENT_DEFS entry

I went with **fallback** rather than a second `id: 'openclaude'` entry because:

1. The issue author explicitly preferred fallback (\"add a detention if `claude` not found then try to find `openclaude`\").
2. OpenClaude is a verbatim Claude Code fork — `buildArgs` / `fallbackModels` / `streamFormat` / `capabilityFlags` would all be duplicated. A new `fallbackBins` slot keeps the Claude config single-sourced and lets future drop-in forks be added with one extra string.
3. The agent picker stays a single \"Claude Code\" entry instead of confusingly splitting into two for what is effectively the same agent.
4. If OpenClaude ever diverges in flags, splitting it back out into its own entry is straightforward (this PR doesn't paint anyone into a corner).

If maintainers prefer the **separate-entry** shape instead, happy to redo — just say the word.

## What changes

[`apps/daemon/src/agents.ts`](https://github.com/nexu-io/open-design/blob/main/apps/daemon/src/agents.ts):

- `AGENT_DEFS[claude]` gains `fallbackBins: ['openclaude']` — a new optional declarative slot. Agents without it (codex, gemini, opencode, cursor-agent, qwen, kiro, hermes) are unchanged.
- New exported helper `resolveAgentExecutable(def)` walks `[def.bin, ...def.fallbackBins]` and returns the first PATH hit (or null). The two existing `resolveOnPath(def.bin)` callsites in `probe()` and `resolveAgentBin()` route through this helper instead of reading `def.bin` directly.
- `stripFns()` drops `fallbackBins` from the API response payload — same posture as `helpArgs` / `capabilityFlags` / probe-only metadata that shouldn't bleed into the public `/api/agents` list.

## Behavior contract

| Setup | Before | After |
|---|---|---|
| Only `claude` on PATH | Detected as Claude Code | Same — primary still wins |
| Both `claude` and `openclaude` on PATH | Detected as Claude Code | Same — `claude` always wins (candidate order matters) |
| Only `openclaude` on PATH | **Not detected** → wrapper required | **Detected as Claude Code**, all chat / model-listing / spawn paths use the resolved `openclaude` binary |
| Neither on PATH | Not detected | Same |
| Other agents (codex / gemini / …) | Detected via `def.bin` | Same — no `fallbackBins` field, unchanged path |

## Tests (5 new, all pass locally)

Added to [`apps/daemon/tests/agents.test.ts`](https://github.com/nexu-io/open-design/blob/main/apps/daemon/tests/agents.test.ts):

1. **Declarative AGENT_DEF guard** — \`claude.fallbackBins\` includes \`'openclaude'\`. Catches silent drift if someone refactors the AGENT_DEF.
2. **Primary wins** — \`resolveAgentExecutable\` returns \`def.bin\` when both candidates exist on PATH (no behavior change for users with both installed).
3. **Fallback engages** — when \`claude\` is missing but \`openclaude\` is on PATH, returns the openclaude path.
4. **All-missing returns null** — when neither candidate is on PATH, returns null (preserves the existing \"not available\" probe outcome).
5. **No-regression for single-bin agents** — defs without \`fallbackBins\` (codex / gemini / opencode / cursor-agent / qwen / kiro / hermes) still resolve through \`def.bin\` exactly as before.

The 4 filesystem-backed cases skip on Windows because \`resolveOnPath\` walks \`PATHEXT\` extensions there and the fixture files don't carry them; the declarative AGENT_DEF assertion runs on every platform.

\`\`\`
Test Files  15 passed (15)
     Tests  230 passed (230)        ← was 225 before this PR
\`\`\`

## Verification

- `pnpm --filter @open-design/daemon test` — 15 files, 230 tests pass (5 new).
- `tsc -p apps/daemon/tsconfig.json --noEmit` — clean.
- `tsc -p apps/daemon/tsconfig.tests.json --noEmit` — clean.
- Built `dist/agents.js` exports the new \`resolveAgentExecutable\` symbol.
- Manual: with `openclaude` symlinked to a no-op script and \`claude\` missing from PATH, \`/api/agents\` reports \`{ id: 'claude', available: true, path: '<path-to-openclaude>' }\` — same shape it would for an authentic `claude` install.

## Out of scope

- Doesn't change OpenClaude's npm-only distribution story (`npm install -g openclaude` still required).
- Doesn't add OpenClaude-specific model fallbacks — Claude Code's existing \`fallbackModels\` (sonnet / opus / haiku / claude-opus-4-5 / …) are reused, which matches what users would expect from a drop-in fork.
- Doesn't add other Claude forks today, but the slot is generic — adding e.g. \`'claude-pro-cli'\` later is a one-string change.

### CODE REVIEW

**Comment by lefarcen** [comment]
Hi @Sid-Qin! 🎉
Thanks for the contribution — great approach to making OpenClaude auto-detect as Claude Code via the fallback slot.
I will run a deep review and get back to you within 24h.

Thanks for making open-design better!
— open-design team

**Review by lefarcen — COMMENTED** [review]
Hey @Sid-Qin! 🎉 Excellent work on this PR — the fallback slot approach is clean, well-tested, and preserves single-source-of-truth for the Claude config. The design choice discussion is thorough and makes the tradeoff clear.

**Code review (Lens A):**

The implementation is solid:
- `resolveAgentExecutable(def)` correctly handles null/undefined/missing `fallbackBins` gracefully
- Existing `resolveOnPath(def.bin)` callsites in `probe()` and `resolveAgentBin()` correctly route through the new helper
- `stripFns()` correctly strips `fallbackBins` from the API response (same posture as `helpArgs` / `capabilityFlags`)
- Test coverage is comprehensive: primary-wins, fallback-engages, all-missing, no-regression for single-bin agents, and declarative AGENT_DEF guard
- The Windows PATHEXT skip with explanation is smart — the declarative test still runs on all platforms

**Reasoning review (Lens B):**

The design choice section is well-argued (fallback vs separate entry), with clear pros/cons. One minor gap:
- **Priority order risk** (minor): If a user has both `claude` and `openclaude` on PATH but `claude` is broken (e.g., corrupted install, missing dependencies), the fallback won't engage because `resolveOnPath('claude')` would still return a path. This is probably acceptable ("installed takes precedence"), but worth documenting in the inline comment at the `fallbackBins` declaration to set expectations for future readers.

P3 suggestion inline below.

Looks good to me overall — deferring final approval to a maintainer.

File: apps/daemon/src/agents.ts [hunk 1]
```diff
@@ -117,6 +117,11 @@ export const AGENT_DEFS = [
     id: 'claude',
     name: 'Claude Code',
     bin: 'claude',
+    // Drop-in forks that ship a CLI argv-compatible with `claude`. Tried in
+    // order if `claude` itself isn't on PATH, so users on a single-binary
+    // install (e.g. only OpenClaude — https://github.com/Gitlawb/openclaude
+    // — issue #235) get auto-detected without writing wrapper scripts.
+    fallbackBins: ['openclaude'],
```
Comment (lefarcen) [inline apps/daemon/src/agents.ts:124]: P3: Consider expanding this comment to set expectations about priority: 'Tried in order **only if `claude` is not on PATH**'. This clarifies that a broken-but-present `claude` install won't fall back to `openclaude` (by design, since resolveOnPath checks existence not validity). Prevents future confusion when users report 'I have both but openclaude isn't used'.

**Review by mrcfps — APPROVED** [review]
Hi @Sid-Qin! 🎉 Thanks for the thoughtful contribution — this looks good to me. The fallback-bin design keeps Claude/OpenClaude behavior single-sourced, preserves the primary \ precedence, and wires the resolved executable consistently through detection and spawn paths. The added tests cover the key edge cases nicely, and I also verified the daemon tests and typecheck locally. 🚀

**Review by mrcfps — APPROVED** [review]
Hi @Sid-Qin! 🎉 Thanks for the thoughtful contribution — this looks good to me. The fallback-bin design keeps Claude/OpenClaude behavior single-sourced, preserves primary Claude precedence, and wires the resolved executable consistently through detection and spawn paths. The added tests cover the key edge cases nicely, and I also verified the daemon tests and typecheck locally. 🚀
